### ISSUE TITLE
Thoughts on using sbt-ci-release?

### ISSUE THREAD

**Issue by gvolpe** [PurpleKingdomGames/tyrian#79]
I wonder if this was considered before? It would be quite useful as every merge to `main` will automatically publish a SNAPSHOT so users don't need to wait for a release to use the latest version :)

**Comment by davesmith00000** [comment]
Definitely should.

You know you hear those stories about plumbers who do brilliant work on the job, but the plumbing in their homes is all stuck together with duct tape? That's me and build systems.

I'll look into it after this release. 😉

**Comment by gvolpe** [comment]
Haha fair enough! I know it takes time to set up, but it's a bleed once you do :) 

BTW you and your work on Tyrian and Indigo had a huge shout-out on [Chris' latest podcast](https://www.tooling-talks.com/), which should come out in a few weeks :)

**Comment by davesmith00000** [comment]
A shout-out on a tooling podcast? ...well now I'm going to be doubly-embarrassed about the ci setup... 😅 😆

**Comment by armanbilge** [comment]
Apologies for sticking my nose in again where it doesn't belong 😛 but you may also want to have a look at [sbt-typelevel](https://typelevel.org/sbt-typelevel/) which in addition to CI release/snapshots sets up MiMa binary-compatibility checking which is pretty important for libraries :)

**Comment by davesmith00000** [comment]
Not at all, the more the merrier @armanbilge. Always good to get another opinion. 😄 

I'll add it to the list of things to do/look into. Tyrian is more framework than library, and I tend to roll everything forward in lock-step (for both Tyrian and Indigo). So as long as you're doing that too when you upgrade versions, you're probably fine, but automated checking certainly couldn't hurt. 👍

**Comment by armanbilge** [comment]
Yes, internally binary compatibility is not so important. But if there are some other libraries that integrate with tyrian/indigo then it would be :)

In any case, it's still good practice to appropriately bump your version when you break binary compatibility.
